test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc

```diff
@@ -599,6 +613,32 @@ TEST_P(RedisProxyIntegrationTest, SimpleRequestAndResponse) {
   simpleRequestAndResponse(makeBulkStringArray({"get", "foo"}), "$3\r\nbar\r\n");
 }
 
+TEST_P(RedisProxyWithCommandStatsIntegrationTest, MGETRequestAndResponse) {
+  initialize();
+  std::string request = makeBulkStringArray({"mget", "foo"});
+  std::string mget_response = "";
+
+  // Make MGET request from downstream
+  IntegrationTcpClientPtr redis_client = makeTcpConnection(lookupPort("redis_proxy"));
+  redis_client->clearData();
+  redis_client->write(request);
+  redis_client->waitForData(mget_response); // MGET returns no response; proxy turns it into GETs
```

Comment:
This is not correct.  The redis_client should receive the response for the mget request.

Reply:
Updated to address this